extensions/grpc/runtime/src/main/java/io/quarkus/grpc/runtime/config/GrpcServerConfiguration.java

```diff
@@ -1,126 +1,277 @@
 package io.quarkus.grpc.runtime.config;
 
+import java.nio.file.Path;
 import java.time.Duration;
+import java.util.List;
 import java.util.Optional;
 import java.util.OptionalInt;
+import java.util.Set;
 
 import io.quarkus.runtime.annotations.ConfigDocSection;
 import io.quarkus.runtime.annotations.ConfigGroup;
-import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.IgnoreProperty;
+import io.smallrye.config.WithDefault;
+import io.vertx.core.http.ClientAuth;
 
-@SuppressWarnings("OptionalUsedAsFieldOrParameterType")
 @ConfigGroup
-public class GrpcServerConfiguration {
+public interface GrpcServerConfiguration {
 
     /**
      * Do we use separate HTTP server to serve gRPC requests.
      * Set this to false if you want to use new Vert.x gRPC support,
      * which uses existing Vert.x HTTP server.
      */
-    @ConfigItem(defaultValue = "true")
-    public boolean useSeparateServer;
+    @WithDefault("true")
+    boolean useSeparateServer();
 
     /**
      * Configure XDS usage, if enabled.
      */
-    @ConfigItem
     @ConfigDocSection(generated = true)
-    public Xds xds;
+    Xds xds();
 
     /**
      * Configure InProcess usage, if enabled.
      */
-    @ConfigItem
-    public InProcess inProcess;
+    InProcess inProcess();
 
     /**
      * The gRPC Server port.
      */
-    @ConfigItem(defaultValue = "9000")
-    public int port;
+    @WithDefault("9000")
+    int port();
 
     /**
      * The gRPC Server port used for tests.
      */
-    @ConfigItem(defaultValue = "9001")
-    public int testPort;
+    @WithDefault("9001")
+    int testPort();
 
     /**
      * The gRPC server host.
      */
-    @ConfigItem(defaultValue = "0.0.0.0")
-    public String host;
+    @WithDefault("0.0.0.0")
+    String host();
 
     /**
      * The gRPC handshake timeout.
      */
-    @ConfigItem
-    public Optional<Duration> handshakeTimeout;
+    Optional<Duration> handshakeTimeout();
 
     /**
      * The max inbound message size in bytes.
      * <p>
      * When using a single server (using {@code quarkus.grpc.server.use-separate-server=false}), the default value is 256KB.
      * When using a separate server (using {@code quarkus.grpc.server.use-separate-server=true}), the default value is 4MB.
      */
-    @ConfigItem
-    public OptionalInt maxInboundMessageSize;
+    OptionalInt maxInboundMessageSize();
 
     /**
      * The max inbound metadata size in bytes
      */
-    @ConfigItem
-    public OptionalInt maxInboundMetadataSize;
+    OptionalInt maxInboundMetadataSize();
 
     /**
      * The SSL/TLS config.
      */
-    public SslServerConfig ssl;
+    SslServerConfig ssl();
 
     /**
      * Disables SSL, and uses plain text instead.
      * If disabled, configure the ssl configuration.
      */
-    @ConfigItem(defaultValue = "true")
-    public boolean plainText;
+    @WithDefault("true")
+    boolean plainText();
+
+    @IgnoreProperty
```

Comment:
I don't think it means what you think it means :)
My bet is that you can keep the method as is without ignoring it: it won't be included.